### DISCUSSION TITLE
Is there anything like "stable RubyGems API"

### DISCUSSION THREAD

**Discussion by voxik** [ruby/rubygems#6455] [General]
Seeing PRs such as #3163, where `Gem.plugins_dir` is replaced by `Gem.plugindir` without any notice, I wonder if there is anything like "stable RubyGems API".

**Comment by deivid-rodriguez** [comment]
#3163 was just a follow up to the rework of the plugins system and happened way before `Gem.plugins_dir` was even released.

Anyways, there's no "official API" but we try not to change public documented methods particularly in `Gem::Specification` since those are commonly used by people.

**Comment by voxik** [comment]
IOW the `Gem.plugins_dir` was never part of any public release, IC. If you considered to rename the method at this stage, it would be likely deprecated first etc. Thx for elaborating.

I am asking, because in the context of #6456, I think the method should be possibly deprecated.

**Reply by deivid-rodriguez** [reply]
Yep, that's correct. And I agree if we were to remove `Gem.plugins_dir` now, it should be deprecated.

**Comment by deivid-rodriguez** [comment]
Closing since I think the original concern was answered.
